ts/components/helpers/withValidatedEmail.tsx

```diff
@@ -0,0 +1,72 @@
+/**
+ * A HOC to display the WrappedComponent when the email is validated, otherwise the RemindEmailValidationOverlay will be displayed
+ */
+
+import React from "react";
+import { View } from "react-native";
+import { NavigationEvents } from "react-navigation";
+import { connect } from "react-redux";
+import { withLightModalContext } from "../../components/helpers/withLightModalContext";
+import RemindEmailValidationOverlay from "../../components/RemindEmailValidationOverlay";
+import { LightModalContextInterface } from "../../components/ui/LightModal";
+import { isEmailEditingAndValidationEnabled } from "../../config";
+import { navigateBack } from "../../store/actions/navigation";
+import { Dispatch } from "../../store/actions/types";
+import { GlobalState } from "../../store/reducers/types";
+import { withConditionalView } from "./withConditionalView";
+
+export type ModalProps = ReturnType<typeof mapDispatchToProps> &
+  LightModalContextInterface;
+
+class ModalRemindEmailValidationOverlay extends React.Component<ModalProps> {
+  constructor(props: ModalProps) {
+    super(props);
+  }
+
+  public componentWillUnmount() {
+    this.props.hideModal();
+  }
+
+  public render() {
+    return (
+      <View>
+        <NavigationEvents
+          onWillBlur={this.props.hideModal}
```

Comment:
Since it is a modal and we don't have `onWillBlur : before losing focus on a tab` is it needed?